src/BatteryStateOfChargeService.cpp

```diff
@@ -43,21 +47,49 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newCurrent_ <= 0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
-}
+    double newTimeSecs = QTime(0,0,0).secsTo(newTime_);
+    double TimeElapsed =(BATTERY_AMP_HOUR_CAPACITY / fabs(newCurrent_)) * SECONDS_TO_HOURS;
+    newTimeSecs = newTimeSecs + TimeElapsed;
+    QTime estimatedTime = QTime(0,0,0).addSecs(newTimeSecs);
 
+    return estimatedTime;
+}
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+    QTime oldTime = QTime();
+    if (oldTime.isNull() == true)
+    {
+        oldTime = QTime(0,0,0);
+
```

Comment:
This section of code seems odd, you set `oldTime` as `QTime()` then check it in a conditional? You must know what the status of `oldTime` is since you set it right before this.